File: python/algo/202501/algo3091.py
# 贪心专题 - 数学贪心 - 基础
class Solution:
    def minOperations(self, k: int) -> int:
        # 自行解答
        ans = 0
        val = cnt = 1
        while val * cnt < k:
            if cnt < val:
                cnt += 1
            else: val += 1
            ans += 1
        return ans

File: cplus/leetcode/202501/algo3091.cpp
#include <bits/stdc++.h>
using namespace std;

// 贪心专题 - 数学贪心 - 基础
class Solution {
public:
    int minOperations(int k) {
        // 自行解答
        int ans = 0, val = 1, cnt = 1;
        while (val * cnt < k) {
            if (cnt < val) cnt ++;
            else val ++;
            ans ++;
        }
        return ans;    
    }
};